supplementary_style_guide/style_guidelines/general-formatting.adoc

```diff
@@ -21,3 +21,72 @@ When the format _day Month year_ causes a presentation or clarity issue, use ins
 
 [[product-names-versions-ref]]
 == Product names and version references
+
+////
+PROPOSED ADDITIONS?
+master-docinfo.xml specific and portal-specific version variables
+URL fragments and composed links?
+correct sequence when combining attributes
+ensure proper sequence for compound attributes
+syntax recommendations: what separators to use? dashes/hyphens, underscores?
+////
+
+Whenever you refer to the name of your product in full, or in its abbreviated form, or when you refer to the major and minor version of your product, avoid using hard-coded references and use attributes instead.
+
+Only use hard-coded version references if the version that you are referring to in that particular case never changes.  
+
+=== Attribute file
+
+Define attributes for product name and product version in a single common shared attributes file for each documentation set or product.
+
+Name the file `attributes.adoc`.
+You may use a different name if necessary, but if you do, ensure that the name of the file indicates to your collaborators that it is an attribute file.
+
+//For a single title, store the file in the same directory as the `master.adoc` file.
+For sharing between multiple titles, store the file in a common directory above the directories that contain your titles, such as `docs/common/` or `docs/topics/templates`.
+
+Include the attributes file at the beginning of the `master.adoc` files of all titles in your documentation set:
+
+.master.adoc
+[source,asciidoc,options="nowrap"]
+----
+\include::common/attributes.adoc[]
+----
+
+TIP: Create a symbolic link pointing to the `common` directory inside the directory containing the `master.adoc` file of your title. It makes including the attribute file in your title easier. +
```

Comment:
This assumes a certain directory structure. I don't think that this is the place to make that assumption. I recall that @sterobin leads a class in setting up directory structure in which this tip would not make sense. Also, I believe that using symlinks to the /common directory is only necessary when a certain Asciidoctor flag is set, which has been the case in the Pantheon toolchain. Not sure that it will continue to be so in Pantheon 2, and I'm also not sure that upstream projects need to worry about this. I know that with oVirt docs, we need to change references to the symlinked /common directory before publishing to upstream.

Reply:
@stoobie You make a good point. I see at least 2 possible solutions here: 
1. Make it clear that what whatever information we provide around directory structure is only an example and primarily relevant to Red Hat projects.
2. Remove all content that deals with the details of directory structure form this chapter, leaving it entirely up to each project's internal rules what directory structure to use.
Any of these seem favorable to you? Would you have another alternative that I have not listed above?

Reply:
@inoxx03 Those are both good ideas. I think it would be helpful to provide a link to the mod docs guide, if that is permissible. I think that the mod docs guide might talk about dir structure.